### ISSUE TITLE
OS X `make launch-kube` step fails

### ISSUE THREAD

**Issue by jnevin** [pachyderm/pachyderm#642]
Following these steps on: https://github.com/pachyderm/pachyderm/blob/master/SETUP.md

On Mac 10.11.5, 
With Docker-For-Mac:
Client:
 Version:      1.12.0-rc3
 API version:  1.24
 Go version:   go1.6.2
 Git commit:   91e29e8
 Built:        Sat Jul  2 00:09:24 2016
 OS/Arch:      darwin/amd64
 Experimental: true

I get this error with `make launch-kube`:

$ make launch-kube
# check that kubectl is installed

which kubectl
/usr/local/bin/kubectl
etc/kube/start-kube-docker.sh
7331b4ba1901d20a5d484855cb1b1b6935cfac3f969df51ba9731e589d2f7e7d
docker: Error response from daemon: Mounts denied:
The path /var/lib/kubelet/
is not shared from OS X and does not belong to the system.
You can configure shared paths from Docker -> Preferences...
..
make: **\* [launch-kube] Error 125

---

Docker Preferences, however, includes the symlinked being sought (/private/var...)

![screen shot 2016-07-11 at 4 29 25 pm](https://cloud.githubusercontent.com/assets/466297/16745635/b1f3f506-4784-11e6-8c3e-4e8b2f59a105.png)

**Comment by jnevin** [comment]
I'm seeing some advice here that might help:
https://forums.docker.com/t/getting-mounts-denied-on-previously-working-container-after-upgrading-to-v1-12-0-beta18-3-gec40b14/17546/5

**Comment by jdoliner** [comment]
Hi @jnevin sorry you hit this.
Kubernetes uses that directory to store it's internal state. I think you just need to add `/var/lib/` to that list of shareable paths and it'll work.

Also if you're looking for the path of least resistance you might want to look into [Minikube](http://kubernetes.io/docs/getting-started-guides/minikube/) I believe it can work with xhyve similar to Docker for Mac and shouldn't have issues like this. I haven't tested it personally though.

**Comment by jmbrunskill** [comment]
I also ran into this issue. I think it could be to do with the way mac's /var folder is actually a symlink to /private/var. My suspicion is that docker is comparing the configured shared path /private/var/lib against the requested path of /var/lib and they don't match. Their UI doesn't let you create a shared folder to the symlink  directly. I worked around this by changing the path in etc/kube/start-kube-docker.sh to a non-symlinked path, however I then ran into other problems with not being able to expose the kubernetes API port. In the end I installed kube-solo which worked perfectly. https://github.com/TheNewNormal/kube-solo-osx
